python/ray/serve/_private/config.py

```diff
@@ -199,6 +204,13 @@ def to_proto_bytes(self):
 
     @classmethod
     def from_proto(cls, proto: DeploymentConfigProto):
+        # MessageToDict from google.protobuf.json_format will automatically convert
+        # bytes into base64 encoded string due to json can't handle bytes directly.
+        # We need to keep in mind for any bytes data in proto, we need to decode it
+        # according.
+        #
+        # See:
+        # https://github.com/protocolbuffers/protobuf/issues/2801#issuecomment-285812695
```

Comment:
@shrekris-anyscale let me know if this comment helps to explain it a bit more or if we should add more content.

Reply:
Alternatively we can just avoid using `MessageToDict`. The purpose of this utility is to convert a proto to a JSON-compatible format, but we don't actually want or need that for any reason here. I think it's just used for convenience to call the constructor at the end of the function.

This seems like it would be simpler/less error prone overall. What do you think?

Reply:
Ok, let me test out how this works 🙂

Reply:
Ended up implementing this `_proto_to_dict()` to convert protobuf into dictionary as there are too many fields to deal 😅